package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -0,0 +1,515 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2020 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+HELANAL --- analysis of protein helices
+=======================================
+
+:Author: Lily Wang
+:Year: 2020
+:Copyright: GNU Public License v3
+
+.. versionadded:: 1.0.0
+
+This module contains code to analyse protein helices using the 
+`HELANAL <http://nucleix.mbu.iisc.ernet.in/helanal/helanal/helanal.html>`_ algorithm
+([Bansal2000]_ , [Sugeta1967]_ ).
+
+`HELANAL <http://nucleix.mbu.iisc.ernet.in/helanal/helanal/helanal.html>`_ 
+quantifies the geometry of helices in proteins on the basis of their Cα
+atoms. It can determine local structural features such as the local helical twist and
+rise, virtual torsion angle, local helix origins and bending angles between
+successive local helix axes. 
+
+Example use
+-----------
+
+You can pass in a single selection::
+
+    import MDAnalysis as mda
+    from MDAnalysis.tests.datafiles import PSF, DCD
+    from MDAnalysis.analysis import helix_analysis as hel
+    u = mda.Universe(PSF, DCD)
+    helanal = hel.HELANAL(u, select='name CA and resnum 161-187')
+    helanal.run()
+
+    print(helanal.summary)
+
+Alternatively, you can analyse several helices at once by passing 
+in multiple selection strings::
+
+    helanal2 = hel.HELANAL(u, select=('name CA and resnum 100-160',
+                                      'name CA and resnum 200-230'))
+
+"""
+from __future__ import division, absolute_import
+
+import warnings
+import numpy as np
+
+import MDAnalysis as mda
+from ..lib import util, mdamath
+from .base import AnalysisBase
+
+
+def pdot(a, b):
```

Comment:
There's some functions in `lib._cutil.pyx` that might do this sort of thing, so maybe either move those there or use them?

Reply:
Is it worth cythonising them?

Reply:
Is it worth cythonising them?